src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1578,6 +1583,99 @@ private static bool HasAnonymousFunctionConversion(BoundExpression source, TypeS
+                elementType = null;
+                return ConstructibleCollectionTypeKind.CollectionInitializer;
+            }
+            else if (isListInterface(compilation, destination, out elementType))
+            {
+                return ConstructibleCollectionTypeKind.ListInterface;
+            }
+
+            elementType = null;
+            return ConstructibleCollectionTypeKind.None;
+
+            static bool isSpanType(CSharpCompilation compilation, TypeSymbol targetType, WellKnownType spanType, [NotNullWhen(true)] out TypeSymbol? elementType)
+            {
+                if (targetType is NamedTypeSymbol { Arity: 1 } namedType
+                    && TypeSymbol.Equals(namedType.OriginalDefinition, compilation.GetWellKnownType(spanType), TypeCompareKind.AllIgnoreOptions))
+                {
+                    elementType = namedType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0].Type;
+                    return true;
+                }
+                elementType = null;
+                return false;
+            }
+
+            static bool implementsIEnumerable(CSharpCompilation compilation, TypeSymbol targetType)
+            {
+                ImmutableArray<NamedTypeSymbol> allInterfaces;
+                switch (targetType.TypeKind)
+                {
+                    case TypeKind.Class:
+                    case TypeKind.Struct:
+                        allInterfaces = targetType.AllInterfacesNoUseSiteDiagnostics;
+                        break;
+                    case TypeKind.TypeParameter:
+                        allInterfaces = ((TypeParameterSymbol)targetType).AllEffectiveInterfacesNoUseSiteDiagnostics;
+                        break;
+                    default:
+                        return false;
+                }
+
+                // This implementation differs from Binder.CollectionInitializerTypeImplementsIEnumerable().
+                // That method checks for an implicit conversion from IEnumerable to the collection type,
+                // but that would allow: Nullable<StructCollection> s = [];
+                // PROTOTYPE: Perhaps adjust the behavior of Binder.CollectionInitializerTypeImplementsIEnumerable()
+                // and use that method instead.
+                var ienumerableType = compilation.GetSpecialType(SpecialType.System_Collections_IEnumerable);
+                return allInterfaces.Any(static (a, b) => a.Equals(b, TypeCompareKind.AllIgnoreOptions), ienumerableType);
+            }
+
+            static bool isListInterface(CSharpCompilation compilation, TypeSymbol targetType, [NotNullWhen(true)] out TypeSymbol? elementType)
+            {
+                if (targetType is NamedTypeSymbol { TypeKind: TypeKind.Interface, Arity: 1 } namedType)
+                {
+                    var definition = namedType.OriginalDefinition;
+                    var listType = compilation.GetWellKnownType(WellKnownType.System_Collections_Generic_List_T);
```

Comment:
- Test where List is missing
- Test where List has no interfaces
- Test where `List<T>` implements `IList<string>`

Reply:
Added `ListInterfaces_NoInterfaces`. The other two are covered in `ListInterfaces_MissingList` and `ListInterfaces_01`.